### PR TITLE
Add error handling in GenerateProvisioningRequest

### PR DESCRIPTION
**Pull request by shenoypritika**
**What this PR does / why we need it**:
- The error returned by GetDbProvRequestAppender was not handled. This could possibly lead to a NPE if it returns nil. This PR handles the error.

**How Has This Been Tested?**:

- Provisioning completed successfully

image

image

### CODE REVIEW

File: ndb_api/db_helpers_test.go [hunk 1]
```diff
@@ -693,3 +693,62 @@ func TestGenerateProvisioningRequest_AgainstDifferentReqData(t *testing.T) {
...
+
+	// Test
+	_, err := GenerateProvisioningRequest(context.Background(), ndb_client, &mockDatabase, reqData)
+	expectedError := errors.New("invalid database type: supported values: mssql, mysql, postgres, mongodb")
+
+	// Assert
+	if err != expectedError && err.Error() != expectedError.Error() {
+		t.Fatalf("expected: %v, got: %v", expectedError, err)
+	}
+}
```
Comment (manavrajvanshi) [inline ndb_api/db_helpers_test.go:754]: Can we merge this test with the reqData test as these are similar, also you can change the test name to something more generic after merging.

**Review by manavrajvanshi — CHANGES_REQUESTED** [review]
Please refactor the UT to fit into the existing test. Thanks!

File: ndb_api/db_helpers_test.go (hunk 1 above)
Comment (shenoypritika) [reply ndb_api/db_helpers_test.go]: Done!

File: config/crd/bases/ndb.nutanix.com_databases.yaml [hunk 2]
```diff
@@ -119,6 +119,7 @@ spec:
                       associated with this database
                     properties:
                       dailySnapshotTime:
+                        default: "12:34:56"
```
Comment (yash-ntnx) [inline config/crd/bases/ndb.nutanix.com_databases.yaml:122]: it is better to have a default snapshot time in the middle of the night when a small DB downtime is easier to tolerate.

Comment (shenoypritika) [reply config/crd/bases/ndb.nutanix.com_databases.yaml]: Sure! Changed it!

**Review by manavrajvanshi — APPROVED** [review]
LGTM, Thanks!

**Review by yash-ntnx — APPROVED** [review]

**Pull request merged**